lutron-config.js

```diff
@@ -31,27 +33,67 @@ module.exports = function (RED) {
             var str = '?OUTPUT,' + devId + ',1';
             this.telnet.getSocket().write(str + '\n');
         };
+
+        const updateStatus = (connected, msg) => {
+            node.statusEvent.emit('update', {
+                fill: connected ? 'green' : 'red',
+                shape: 'dot',
+                text: msg
+            });
+        };
+
+        const reconnect = () => {
+            // Try reconnecting in 1 minute
+            node.log('reconnecting telnet')
+            setTimeout(() => this.telnet.connect(params), 60000);
+        }
+
+        // Telnet handlers
         this.telnet.on('data', (function (self, pkt) {
             self.lutronRecv(pkt);
         }).bind(null, node));
         this.telnet.on('connect', function () {
-            this.connected = true;
-            console.log('telnet connect');
+            node.connected = true;
+            node.log('telnet connect');
+            updateStatus(true, 'connected');
         });
         this.telnet.on('close', function () {
-            this.connected = false;
-            console.log('telnet close');
+            if (node.connected) {
+                node.log('telnet close');
+            }
+            node.connected = false;
+            updateStatus(false, 'closed');
         });
         this.telnet.on('error', function () {
-            console.log('telent error');
+            if (node.connected) {
+                node.warn('telnet error');
+            }
+            updateStatus(false, 'telnet error');
         });
         this.telnet.on('failedlogin', function () {
-            console.log('telent failed login');
+            node.warn('telnet failed login');
+            updateStatus(false, 'login failed');
         });
+        this.telnet.on('timeout', function () {
+            if (node.connected) {
+                node.log('telnet timeout');
+            }
+        });
+        this.telnet.on('end', function () {
+            if (node.connected) {
+                // This happens periodically (on bridge updates?)
+                // so try reconnecting afterwards
+                node.warn('telnet remote ended connection');
+                reconnect();
+            }
+            updateStatus(false, 'ended');
```

Comment:
It might be nice to have the status say 'reconnecting' if it's trying to reconnect.

Reply:
That makes sense, I'll update